pkg/graveler/uncommitted_diff_iterator.go

```diff
@@ -28,59 +29,69 @@ func NewUncommittedDiffIterator(ctx context.Context, manager CommittedManager, l
 	}
 }
 
-func (d *uncommittedDiffIterator) valueExistsInCommitted(val ValueRecord) (bool, error) {
+func (d *uncommittedDiffIterator) getValueFromCommittedIfExists(val ValueRecord) (*Value, error) {
```

Comment:
@guy-har @arielshaqed A little late to the party, but I still want to raise a little concern:
For each value that you read from the committed data, you go thru the cycle of fetching metarange, reading it, going thru and finding the matching range, fetching the range and searching it. This could happen multiple times for a single range, if some uncommitted values fell into the same range.
Instead you could list the commit, and `SeekGE` to the values. It will save you some overhead of re-handling the same metarange and ranges over and over again since the committed iterator has a state.

Having said all that, I know that we cache metaranges/ranges and I assume that most installations under a normal workload and a decent disk size, will probably read these files from cache, therefore saving 99.9% of the above overhead (i.e. reading from s3).

Reply:
Thanks, Great Input.
Don't think it should be part of this PR anyway.
Created a new another PR for it.
#1865